server/src/chunk_logic.rs

```diff
@@ -84,9 +86,11 @@ impl<'a> System<'a> for ChunkLoadSystem {
         use specs::prelude::SystemData;
 
         let generator = world.fetch_mut::<Arc<dyn WorldGenerator>>().clone();
+        let config = world.fetch_mut::<Arc<Config>>().clone();
```

Comment:
Rather than fetching and cloning the `Arc`, I would recommend just cloning `config.world.name`.